Sources/MessagingInApp/State/InAppMessageReducer.swift

```diff
@@ -49,16 +49,32 @@ private func reducer(action: InAppMessageAction, state: InAppMessageState) -> In
 
     case .displayMessage(let message):
         if let queueId = message.queueId {
+            // If the message should be tracked shown when it is displayed, add the queueId to shownMessageQueueIds.
+            let shownMessageQueueIds = action.shouldTrackMessageShown
+                ? state.shownMessageQueueIds.union([queueId])
+                : state.shownMessageQueueIds
+
             return state.copy(
                 currentMessageState: .displayed(message: message),
                 messagesInQueue: state.messagesInQueue.filter { $0.queueId != queueId },
-                shownMessageQueueIds: state.shownMessageQueueIds.union([queueId])
+                shownMessageQueueIds: shownMessageQueueIds
             )
         }
         return state
 
     case .dismissMessage(let message, _, _):
-        return state.copy(currentMessageState: .dismissed(message: message))
+        let shownMessageQueueIds: Set<String>
```

Comment:
could be initialized with `state.shownMessageQueueIds`

Reply:
Updated